main.go

```diff
@@ -133,6 +133,22 @@ func (d *Downloader) downloadFileWithTimeout(userCtx context.Context, u string)
 	}
 }
 
+func (d *Downloader) getContentSizeHeader(ctx context.Context, u string) (uint64, error) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodHead, u, nil)
```

Comment:
Question here: why _with context_ if we're not using the context?

@danielfireman might help. Is there some code missing? Maybe we can cancel the execution of the function on `ctx.Done()`, right?

Reply:
I thought the same thing actually. I was intending to remove this ctx, but I wasn't sure if it could be removed.

Reply:
The passed-in `ctx` is going to rule the HTTP request. The following text fragment from the function documentation might help understanding.

> For an outgoing client request, the context controls the entire lifetime of a request and its response: obtaining a connection, sending the request, and reading the response headers and body. See the Request type's documentation for the difference between inbound and outbound request fields.